File: freight-triage-polyglot-ledger-recovery/environment/intake/src/com/freight/norm/StripNonAlnum.java
package com.freight.norm;

/** Normalizer: strip non alnum. */
public final class StripNonAlnum implements Normalizer {

    @Override
    public String name() {
        return "strip_non_alnum";
    }

    @Override
    public String apply(String text) {
        StringBuilder out = new StringBuilder(text.length());
        for (int i = 0; i < text.length(); i++) {
            char c = text.charAt(i);
            if ((c >= '0' && c <= '9') || (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z')) {
                out.append(c);
            }
        }
        return out.toString();
    }
}

File: freight-triage-polyglot-ledger-recovery/environment/reconcile/internal/normx/strip_non_alnum.go
package normx

// StripNonAlnum normalizes with the strip non alnum rule.
func StripNonAlnum(text string) string {
	out := make([]byte, 0, len(text))
	for i := 0; i < len(text); i++ {
		c := text[i]
		if (c >= '0' && c <= '9') || (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') {
			out = append(out, c)
		}
	}
	return string(out)
}
